### ISSUE TITLE
Organize notes into notebooks

### ISSUE THREAD

**Issue by dexter0195** [xournalpp/xournalpp#799]
It would be great to organize notes into notebooks or folders from inside the app as it is done in OneNote.
It's true that you can navigate the filesystem from a file explorer and then select the file to edit but the feeling is not the same.
Furthermore you could make hyperlinks between notes with that.

**Comment by amirji** [comment]
It would be great if there would be such that.

**Comment by veecue** [comment]
I will be working on such a feature. I was thinking about another Document type called a xournalpp workspace that organizes .xopp files on two directory levels.
Those would be displayed nicely in the sidebar similar to OneNote

**Comment by Erater** [comment]
Are there any Updates on this feature?

**Comment by dibbel26** [comment]
I would really enjoy such a feature, as you said, it might be possible to stick to the current file format but with some sort of "One Note like" notebook organisation in which case the existing folder structure could be interpreted as such notebooks.

**Comment by Fatih20** [comment]
> I would really enjoy such a feature, as you said, it might be possible to stick to the current file format but with some sort of "One Note like" notebook organisation in which case the existing folder structure could be interpreted as such notebooks.

Yes. User could just setup a special folder that Xournal then can use to make folders within it to organize the notebooks.

**Comment by Fatih20** [comment]
@veecue Any update?

**Comment by veecue** [comment]
Nothing so far. Just haven't had the time so far. Feel free to implement it ;). The easiest possibility would be to allow opening folders and just have a sidebar alternative that displays the opened folder in a tree view.

**Comment by rolandlo** [comment]
@veecue Your ideas sound good to me. I think this would really make a big impact. Really hope you will find time to code it up.

**Comment by rolandlo** [comment]
Having discovered [Joplin](https://joplinapp.org/) for managing notebooks and its ability to attach all kind of files (including xopp-files), I am not so sure anymore whether this is really something we should implement in Xournal++. Checkout [this blog post](https://linux-in-der-schule.de/2020/11/) and in particular the video near the word Xournal++. It looks great to me. Even syncing up to Nextcloud (and alike) is supported. I will organize my stuff in that way in future, I think. That's kind of the missing piece I needed to organize my workflow.

**Comment by mizhozan** [comment]
@rolandlo Even without using Joplin, it's possible to just use your favorite FS tools organized in files and folders as you wish. This way xournalpp focuses on what matters: creating and enriching note taking experience for one note and organizing different notes is the job of file system. This is my 2cent as I really found the POSIX way, the most efficient way. We need a very good note taking tool and searching through those notes, organizing them and etc can be done by their appropriate tools. For example for searching I just use my `rg` or even use vim here and there, not to mention favorite file manager organizing the notes. This being said, having for example the capability to follow links/hyperlinks or being able to launch (e.g. open a PDF link) via xdg-open would be very nice in further organization than having basically built-in file-manager inside.

**Comment by olbotta** [comment]
As @rolandlo pointed out, joplin is a really nice piece of software. i have been using it for a few months now, and coupling it with xournal++ keeps me away from onenote and windows. However there is no way to visualize the note directly inside joplin.
I believe that a joplin plugin to integrate an image-like visualization for .xopp files would be marvelous. The editing would still be made in xournal, but having the ability to include hand-written notes would bring hybrid note-taking to the next level.

workaround: export the xournal++ note to pdf and include that in joplin. (i know, this is quite inconvenient...)

**Comment by AmritPant** [comment]
Any Update on this Feature?
